src/components/ellenTicTacToe.tsx

```diff
@@ -0,0 +1,240 @@
+              clickedSquare[1]
+            })`
+          : 'Go to game start';
+        return (
+          <li key={move}>
+            <button
+              className={css(
+                {
+                  font: 'inherit',
+                  margin: 3,
+                  overflow: 'visible',
+                  textTransform: 'none',
+                },
+                { ':focus': { fontWeight: 'bold' } },
+              )}
+              onClick={() => this.jumpTo(move)}
+            >
+              {desc}
+            </button>
+          </li>
+        );
+      },
+    );
+    let status;
+    if (winner) {
+      status = 'Winner: ' + winner[0];
+    } else if (current.squares.includes(null) === false) {
+      status = 'Tie: No available moves remaining';
+    } else {
+      status = 'Next player: ' + (this.state.xIsNext ? 'X' : 'O');
+    }
+    return (
+      <div
+        className={css({
+          display: 'flex',
+          flexDirection: 'row',
+        })}
+      >
+        <div className="game-board">
+          <Board
+            squares={current.squares}
+            onClick={i => this.handleClick(i)}
+            winningArray={createWinningArray(
+              calculateWinner(history[this.state.stepNumber].squares),
+            )}
+          />
+        </div>
+        <div className={css({ 'margin-left': '20px' })}>
+          <div>{status}</div>
+          <div
+            className={css({
+              float: 'right',
+              marginBottom: '20px',
+              marginLeft: '300px',
+            })}
+          >
+            <ol
+              className={css({
+                display: 'flex',
+                flexDirection: 'column',
+                listStyleImage: 'none',
+                listStylePosition: 'outside',
+                marginBottom: '1.45rem',
+                marginLeft: '1.45rem',
+                marginRight: 0,
+                marginTop: 0,
+                padding: 0,
+              })}
+            >
+              {moves}
+            </ol>
+            &ensp;&ensp;Toggle list order &ensp;
+            <input type="checkbox" id="toggle" className="toggle" />
+          </div>
+          {/* <button onClick={() => reverseOrder(moves)}>
+            Toggle list order
+          </button> */}
+        </div>
+      </div>
+    );
+  }
+}
+// ========================================
+type Line = [number, number, number];
+function calculateWinner(squares: string[]): [string, Line] {
+  const lines: Line[] = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+  const winningLine = _.find(
+    lines,
+    ([a, b, c]) =>
+      squares[a] && squares[a] === squares[b] && squares[a] === squares[c],
+  );
+  return winningLine ? [squares[winningLine[0]], winningLine] : null;
+}
+function createWinningArray(winningSquares: [string, Line] | null): string[] {

```

Comment:
It's a bit messy taking the tuple here when we only care about the second part; I'd say it's better to change the place this is used to `createWinningArray(_.get(calculateWinner(squares), 1))` and have this just be `createWinningArray(winningSquares: Line | null)`.